code/modules/mob/dead/observer/observer.dm

```diff
@@ -304,6 +304,12 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		qdel(src)
 		return
 
+	if (ishivemindmember(src))
```

Comment:
EEEEEEEEEh this really should be an event `onGhosting` with added listeners on the hivemind side of things, and something similar ought to be done for the other parts of this porc, but that's not critical